compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1922,4 +1922,36 @@ object messages {
     val msg = hl"Traits cannot redefine final $method from ${"class AnyRef"}."
     val explanation = ""
   }
+
+  case class UnableToExtendSealedClass(pclazz: Symbol)(implicit ctx: Context) extends Message(UnableToExtendSealedClassID) {
+    val kind = "Syntax"
+    val msg = hl"cannot extend sealed $pclazz in different compilation unit"
```

Comment:
I would refactor that and the extended message to only talk about files instead. It seams unnecessary to talk about compilation units here.

Here is what we have in the spec if that helps:

> The sealed modifier applies to class definitions. A sealed class may not be directly inherited, except if the inheriting template is defined in the same source file as the inherited class. However, subclasses of a sealed class can be inherited anywhere.